### PR TITLE
Splash Class

### PR DESCRIPTION
**Pull request by SeyedMojtaba1**

### CODE REVIEW

File: src/providers/splash.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        }
+      );
+      return response.json();
+    } catch (error) {
+      console.error("Error fetching order book:", error);
```
Comment (mkermani144) [inline src/providers/splash.ts:25]: It's not a good practice to catch an error only to log it, and then re-throw it. You can simply remove the `try/catch` block.

File: src/providers/splash.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+class Splash implements Provider {
+  private baseUrl: string;
+
+  constructor(url: string) {
+    this.baseUrl = url;
+  }
+
+  async getOrderBook(marketId: string) {
+    try {
+      const response = await fetch(
```
Comment (mkermani144) [inline src/providers/splash.ts:14]: The type of this variable is implicitly `any`, which is not a good thing. You should explicitly define the type for this variable based on the structure of the response.

File: src/providers/splash.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  }
+
+  async x2y(marketId: string, amount: AmountWithDecimal) {
+    const orderBook = await this.getOrderBook(marketId);
+
+    const ask = orderBook.asks[0].price;
+
+    return {
+      amount: ask.toFixed(amount.decimals),
```
Comment (mkermani144) [inline src/providers/splash.ts:36]: There are two important issues here:
1. When we say token X has amount of `1000` with decimal of `2`, it means the human readable format will be `10.00`, which is simply `10`.
2. `amount.decimals` is the decimals for input value. The returned value should have its own decimals, based on the token. For example, ADA has a decimal of 6, but ERG a decimal of 9. It seems that Splash uses the following endpoint for getting token data:
https://spectrum.fi/cardano-token-list-v2.json

**Review by mkermani144 — CHANGES_REQUESTED** [review]
In addition, your lock file has changed, without any package being installed. I think it's the result of installing `axios` in your previous commit, and should be reverted.

File: src/providers/splash.ts (hunk 1 above)
Comment (SeyedMojtaba1) [reply src/providers/splash.ts:25]: Ok. I delete it.

File: src/providers/splash.ts (hunk 2 above)
Comment (SeyedMojtaba1) [reply src/providers/splash.ts:14]: I fixed it.

File: src/providers/splash.ts [hunk 4]
```diff
@@ -2,51 +2,67 @@ import { AmountWithDecimal, Provider } from "@/types/core";
...
   async x2y(marketId: string, amount: AmountWithDecimal) {
     const orderBook = await this.getOrderBook(marketId);
 
     const ask = orderBook.asks[0].price;
 
-    return {
-      amount: ask.toFixed(amount.decimals),
-      decimals: amount.decimals,
-    };
+    return;
```
Comment (mkermani144) [inline src/providers/splash.ts:55]: I think you meant `return Number(ask)`. Same for `y2x`.

File: src/providers/splash.ts [hunk 5]
```diff
@@ -2,51 +2,67 @@ import { AmountWithDecimal, Provider } from "@/types/core";
...
+        headers: {
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    const data = await response.json();
+    return data as OrderBookResponse;
   }
 
   async x2y(marketId: string, amount: AmountWithDecimal) {
```
Comment (mkermani144) [inline src/providers/splash.ts:50]: As I mentioned, `amount` is of type `number`, and not `AmountWithDecimal` anymore. Make sure function parameters and return types match those of `Provider` interface itself.

**Review by mkermani144 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**